AR-Framework/src/at/fhv/itb6/arp/inputInterface/InputDetection.java

```diff
@@ -31,7 +34,7 @@ public InputDetection(InputConfiguration inputConfiguration){
      * @throws GamebordersNotDetectedException
      * @throws NoMarkerDetectedException
      */
-    public InputAction getUserInput() throws GamebordersNotDetectedException, NoMarkerDetectedException {
+    public InputAction getUserInput(CursorStatusListener callback) throws GamebordersNotDetectedException, NoMarkerDetectedException {
```

Comment:
Since the method of getting a user input can take quite some time, the callback is used to update the UI to show the cursor position / progress of a "click". The returned InputAction is seen as a "click".
Returning a game position seems a little high level to me. With the current concept it would be easily possible to implement more complex UIs where you could for example navigate through menus etc.